### ISSUE TITLE
Incorrect debug information leading to bad visualzization for &str and slice arguments in MSVC debuggers.

### ISSUE THREAD

**Issue by nanguye2496** [rust-lang/rust#81894]
(Fix for this problem: https://github.com/rust-lang/rust/pull/81898)

When lowering MIR to LLVM IR, the compiler decomposes every function argument of type &str or slice into two components: a pointer to the memory address referenced to by the &str or slice and an integer represent the length of the str or slice. Then, the original argument is constructed in the body of the function from the data pointer and the integer arguments. Since the original argument is declared in the body of the function that owns it, it should be marked as a LocalVariable instead of an ArgumentVariable. This confusion makes MSVC debuggers unable to visualize &str and slice arguments correctly. Because these debuggers implicit treat arguments whose sizes are greater than that of a pointer as pass by reference, the values of &str or slice arguments are treated as their reference, leading to incorrect visualization for &str and slice arguments. 

I tried this code:

```rust
#[inline(never)]
pub fn nam_foo_function(first_var: &str) -> usize {
    first_var.len()
}

fn main() {
    s = "hello";
    nam_foo_function(&s[1..3]);
}
```

When stepping through the code in VS Code C++ debugger, I expected to see this happen:
![image](https://user-images.githubusercontent.com/74681961/107279546-07538d00-6a0c-11eb-8ff1-26fb19dc2828.png)

Instead, this happened:
![image](https://user-images.githubusercontent.com/74681961/107279588-15a1a900-6a0c-11eb-8ff9-d4ecf013c4da.png)

### Meta
This error is found on rust 1.51.0-dev, 1.50, and 1.49.

**Comment by nanguye2496** [comment]
I made a PR for this issue:
https://github.com/rust-lang/rust/pull/81898

**Comment by rylev** [comment]
@nanguye2496 Can this be closed now that #81898 has been merged?

**Comment by nanguye2496** [comment]
Fix https://github.com/rust-lang/rust/pull/81898  has been merged.
